### PR TITLE
GHA - external dependency testing workflow: add a schedule to run once every week

### PR DESCRIPTION
**Pull request by AjayThorve**
Based on the discussion in #452, adding a weekly schedule to run the external dependency testing workflow.

cc @exactlyallan @ajschmidt8

### CODE REVIEW

File: .github/workflows/test-external.yaml
Comment (ajschmidt8) [inline .github/workflows/test-external.yaml:1]: the last line of this file references input values:

```
${{ inputs.external-project }}
```

The inputs object will be empty when this job is triggered via a cron event.

Therefore, you'll need to provide a default value when the inputs are empty.

Something like this:

```
${{ inputs.external-project || 'default_value' }}
```

The string value must be wrapped in single quotes, not double quotes.

The doc page below has additional information on expressions in GitHub Actions for reference:

- https://docs.github.com/en/actions/learn-github-actions/expressions

Comment (AjayThorve) [reply .github/workflows/test-external.yaml:1]: updated it to have a default value of 'all', thanks for the helpful link. The second input parameter is not required for the cron job, and hence left it as is. Let me know if this works

**Comment by AjayThorve** [comment]
test failures seem unrelated to the changes in this PR, investigating now.

**Review by ajschmidt8 — COMMENTED** [review]
Last comment:

Should we update these RAPIDS versions to the latest stable? `23.04`?

- https://github.com/rapidsai/cuxfilter/blob/branch-23.06/ci/test_external.sh#L10

I assume we want them to test against stable versions and not nightlies, right?

If we were testing against nightlies, we could add an entry to https://github.com/rapidsai/cuxfilter/blob/branch-23.06/ci/release/update-version.sh to keep them up-to-date.

**Comment by AjayThorve** [comment]
> If we were testing against nightlies, we could add an entry to https://github.com/rapidsai/cuxfilter/blob/branch-23.06/ci/release/update-version.sh to keep them up-to-date.

Great idea, thanks! Pushed the changes

**Review by ajschmidt8 — APPROVED** [review]

**Pull request merged**